### ISSUE TITLE
Support in-repo addons

### ISSUE THREAD

**Issue by tomdale** [embroider-build/embroider#11]
Currently, I believe only addons installed in `node_modules` get transformed correctly from V1 to V2. We should support in-repo addons as well, that live in `lib/`.

From @ef4:

> we would need to discover that they exist and compile them to v2 format, same as other addons. Then in the output of the phase 2 build, we should put them into the app's devDependencies list and add symlinks, so that the final stage packager can treat them as regular NPM packages.
> probably we can use one of the package.json features like 
```json
devDependencies: {
  "your-in-repo-addon": "file:./lib/your-in-repo-addon"
}
```
> we would still need to make the symlink, because NPM would have done that at NPM install time
> they will now be able to conflict with NPM package names, but that was already kinda true, since if you had an addon directory in an in-repo addon, it's already fighting with other true NPM addons over the package name;
> if we find real conflicts (where somebody already has a dep in package.json with the same name), we can even rename them. There's already machinery for remapping imports in your app code, to follow the rename.

**Comment by lifeart** [comment]
can I add `yarn workspaces` support into this issue?
for projects like this:
```
- root-folder
-- node_modules
-- ember-app
-- addon-one
-- addon-two
.yarn-lock
```

**Comment by ef4** [comment]
I think we already support yarn workspaces.

**Comment by ef4** [comment]
This was fixed as of  at least 0.3.0.
